test/README.md

```diff
@@ -1,95 +1,296 @@
+# key), server.csr (the certificate signing request) and, server.crt (the certificate)
+~~~
+
+###### Setup native basic auth
+~~~ sh
+sudo apt-get install apache2-utils
+# Create password file
+: | sudo tee htpasswd
+# Create username and password. Replace the username and password strings 
+# below with a user/password combo that you like
+echo "password" | sudo htpasswd -iB htpasswd username
+~~~
+
+###### Configure and run the local docker repository
+~~~ sh
+# Set up the docker registry to listen on port 5000 and to restart automatically.
+sudo docker run -d  -p 5000:5000 \
+  --name registry  \
+  -v /home/azureuser/auth/:/etc/security \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/etc/security/server.crt  \
+  -e REGISTRY_HTTP_TLS_KEY=/etc/security/server.key \
+  -e REGISTRY_AUTH=htpasswd   \
+  -e REGISTRY_AUTH_HTPASSWD_PATH=/etc/security/htpasswd   \
+  -e REGISTRY_AUTH_HTPASSWD_REALM="Registry Realm"  \
+  --restart always  \
+  registry:2
+
+# Run docker ps to validate that the registry came up successfully
+sudo docker ps
+# If all goes well, you will see the something like below when you run Docker ps.
+# 59867fe47a5b   registry:2   "/entrypoint.sh /etc…"   6 seconds ago   Up 6 seconds   0.0.0.0:5000->5000/tcp, :::5000->5000/tcp   registry
+
+# Run docker login to validate setup. This step will cache your credentials on the host
+# and will be required later when you build and push docker images
+~~~
+
+##### Install .NET
+End-to-end tests are written in .NET Core and run with the `dotnet test` command, so you need to [install .NET Core SDK](https://docs.microsoft.com/en-us/dotnet/core/install/sdk). *Note: You will have to install .NET version 3.1*
+
+#### Cloud side setup for the tests
+##### IoT Hub
+If you don't already have an existing IoTHub, create one. There is no special configuration required, except for making sure that your IoTHub is enabled for public access.
+
+##### Device Provisioning Service (DPS)
+Create a DPS instance. A subset of the end to end tests will use DPS for testing device provisioning scenarios. 
+###### Create enrollment group for symmetric key based enrollment
+In your DPS instance, create new enrollment group. For this enrollment group, set the attestation type to be symmetric key, toggle IoT Edge Device to true, select your iothub by clicking on Link a new Iot Hub, and select the access policy to be iotHubOwner
+		
+###### Create enrollment group for X.509 certificate based enrollment
+In your DPS instance, create another enrollment group for X.509 certificate based enrollment. Link this new group to your IotHub, associate it with the your machine's root CA certificate and set the IotEdge Device toggle to true.
+
+##### Create a storage account/container
+Create a storage container, with access level set to private. Create a SAS token URL ( allow https only, permissions set to "write" only). Note down this URL as you will need it to set the enviornment.

```

Comment:
In the DPS section above, you wrote:

> A subset of the end to end tests will use DPS for testing device provisioning scenarios.

It would be good to have a similar explanation here for the storage account, since you don't need it at all if you skip the `TestUploadModuleLogs` and `TestUploadSupportBundle` tests.

Reply:
Good idea.